src/erlcloud_aws.erl

```diff
@@ -494,3 +501,189 @@ get_filtered_statuses(ServiceNames, Statuses) ->
                 ServiceNames)
         end,
         Statuses).
+
+
+%%%---------------------------------------------------------------------------
+-spec profile() -> {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
+%% @doc Retrieve a config based on default credentials
+%%
+%% This function will retrieve the credentials for the <em>default</em>
+%% profile, in the same way as the AWS CLI tools, and construct a {@link
+%% #aws_config{}. config record} to be used accessing services.
+%%
+%% @see profile/2
+%%
+profile() ->
+    profile( default, [] ).
+
+
+%%%---------------------------------------------------------------------------
+-spec profile( Name :: atom() ) -> {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
+%% @doc Retrieve a config based on named credentials profile
+%%
+%% This function will retrieve the credentials for the named profile, in the
+%% same way as the AWS CLI tools, and construct a {@link
+%% #aws_config{}. config record} to be used accessing services.
+%%
+%% @see profile/2
+%%
+profile( Name ) ->
+    profile( Name, [] ).
+
+
+-type profile_option() :: {role_session_name, string()}
+                          | {role_session_secs, 900..3600}.
+
+%%%---------------------------------------------------------------------------
+-spec profile( Name :: atom(), Options :: [profile_option()] ) ->
+                     {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
+%% @doc Retrieve a config based on named credentials profile
+%%
+%% This function will read the <code>$HOME/.aws/credentials</code> file used
+%% by the AWS CLI tools and construct a {@link #aws_config{}. config record}
+%% to be used accessing services for the named profile.  This supports both
+%% direct credentials that appear in the profile:
+%%
+%% <code><pre>
+%%  [default]
+%%  aws_access_key_id = XXXXXXXXXXXXXXXXXXX2
+%%  aws_secret_access_key = yyyyyyyyyyyyyyyyyyyyyyyyyy+yyyy/yyyyyyyy2
+%% </pre></code>
+%%
+%% as well as indirection using <em>source_profile</em>:
+%%
+%% <code><pre>
+%%  [foo]
+%%  source_profile = default
+%% </pre></code>
+%%
+%% and finally, will supports the <em>role_arn</em> specification, and will
```

Comment:
role based credential will expire in `session_secs`. Profile loading assumes role only once. 
As a library consumer its unclear (since expiration is not stored in `#aws_config`) when and how I should re-assume role before expiration before execution next AWS call.

I see two options:
- do not assume the role while loading profile and let consumers do it the way they wish
- provide a facility which will hide the token renewal from the library consumer. Preferable option but one may argue it's library complication.